include/uapi/sound/sof-ipc.h

```diff
@@ -278,7 +284,9 @@ struct sof_ipc_dai_ssp_params {
 /* HDA Configuration Request - SOF_IPC_DAI_HDA_CONFIG */
 struct sof_ipc_dai_hda_params {
 	struct sof_ipc_hdr hdr;
-	/* TODO */
+	u32 hda_count;
+	u32 hda_comp_id[SOF_IPC_MAX_CHANNELS];
+	u32 hda_dma_ch[SOF_IPC_MAX_CHANNELS];
 } __attribute__((packed));
```

Comment:
what is the use of this hda_params structure?
I really don't understand how this might work long term, e.g if we have two host-DSP transfers (low-latency and  media) which get mixed on the same output? Are we going to send the same hda_dma_ch values twice, possibly after the link DMA is already running to deal with the first stream?
The problem for me is that you stick the hdaudio params in the IPC stream params when they should be in this

````text 
 /* HDA Configuration Request - SOF_IPC_DAI_HDA_CONFIG */
struct sof_ipc_dai_hda_params {
	struct sof_ipc_hdr hdr;
	/* TODO */
} __attribute__((packed));
````
Same if we add support for tone generation, you would need to pass the link DMA config *WITHOUT* any ipc for streams

Reply:
Hi Pierre, your questions are discussed many times and no conclusion is gotten.
For question one, @lgirdwood @xiulipan @tlauda can you help to make it clear.
For tone, @lgirdwood  @ranj063 , can you help to make it clear.
Thanks!